inc/Ui/Revisions.php

```diff
@@ -0,0 +1,330 @@
+                    'alt'    => '',
+                ]);
+                $form->addHTML($display_name);
+            }
+            $form->addHTML(' ');
+
+            if ($media_id) $form->addTagOpen('div');
+
+            if ($info['sum']) {
+                $form->addTagOpen('span')->addClass('sum');
+                if (!$media_id) $form->addHTML(' – ');
+                $form->addHTML('<bdi>'. hsc($info['sum']) .'</bdi>');
+                $form->addTagClose('span');
+            }
+            $form->addHTML(' ');
+
+            $form->addTagOpen('span')->addClass('user');
+            if ($info['user']) {
+                $form->addHTML('<bdi>'. editorinfo($info['user']) .'</bdi>');
+                if (auth_ismanager()) {
+                    $form->addHTML(' <bdo dir="ltr">('. $info['ip'] .')</bdo>');
+                }
+            } else {
+                $form->addHTML('<bdo dir="ltr">' .$info['ip'] .'</bdo>');
+            }
+            $form->addTagClose('span');
+            $form->addHTML(' ');
+
+            $form->addHTML(html_sizechange($info['sizechange']));
+
+            if ($media_id) $form->addTagClose('div');
+
+            $form->addTagClose('div');
+            $form->addTagClose('li');
+        }
+
+        // end of revision list
+        $form->addTagClose('ul');
+
+        // show button for diff view
+        if (!$media_id) {
+            $form->addButton('do[diff]', $lang['diff2'])->attr('type', 'submit');
+        } else {
+            $form->setHiddenField('mediado', 'diff');
+            $form->addButton('', $lang['diff2'])->attr('type', 'submit');
+        }
+
+        $form->addTagClose('div'); // close div class=no
+
+        // emit HTML_REVISIONSFORM_OUTPUT event
+        Event::createAndTrigger('HTML_REVISIONSFORM_OUTPUT', $form, null, false);
+        print $form->toHTML();
+
+        print DOKU_LF;
+
+        // provide navigation for pagenated revision list (of pages and/or media files)
+        print '<div class="pagenav">';
+        $last = $first + $conf['recent'];
+        if ($first > 0) {
+            $first = $first - $conf['recent'];
+            if ($first < 0) $first = 0;
+            print '<div class="pagenav-prev">';
+            if ($media_id) {
+                print html_btn('newer',$media_id,"p",media_managerURL(array('first' => $first), '&amp;', false, true));
+            } else {
+                print html_btn('newer',$id,"p",array('do' => 'revisions', 'first' => $first));
+            }
+            print '</div>';
+        }
+        if ($hasNext) {
+            print '<div class="pagenav-next">';
+            if ($media_id) {
+                print html_btn('older',$media_id,"n",media_managerURL(array('first' => $last), '&amp;', false, true));
+            } else {
+                print html_btn('older',$id,"n",array('do' => 'revisions', 'first' => $last));
+            }
+            print '</div>';
+        }
+        print '</div>';
+
+        print DOKU_LF;
+    }
```

Comment:
This function is suuuper long and should be split up.

Reply:
fixed in 67ef3e8. I hope the code is easy to understand, descriptive whole sequence of the Ui\Revision::show() method.